core/trace/dt_span_exporter.go

```diff
@@ -138,9 +125,37 @@ func (e *dtSpanExporterImpl) export(ctx context.Context, t exportType, spans dtS
 		return errors.New("unexpected response code: " + strconv.Itoa(resp.StatusCode))
 	}
 
+	e.logger.Debug("Export successful")
 	return nil
 }
 
+func (e *dtSpanExporterImpl) exportWithRetry(ctx context.Context, t exportType, serializedSpans []byte) (*http.Response, error) {
+	reqBody := bytes.NewBuffer(serializedSpans)
+	req, err := e.newRequest(ctx, reqBody)
+	if err != nil {
+		return nil, err
+	}
+	exportStartTime := time.Now()
+	resp, err := e.performHttpRequest(req, t)
+	exportDuration := time.Since(exportStartTime)
+
+	// Retry once if the connection was reset by the backend and the request took less than cConnectionResetTimeLimit.
+	if errors.Is(err, io.EOF) {
```

Comment:
Could we be more conservative here and only retry if the connection was actually reset and not if we got an EOF for different reasons? Altough I'm not sure if that actually makes sense since span export *should* be idempotent (or fixed to be so on the server if necessary) anyways.